### PR TITLE
autoPatchelfHook: fix bug introduced by #101142

### PR DESCRIPTION
**Pull request by DavHau**
Fixes breakage intrduced by #101142

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by eadwu — APPROVED** [review]
Fixes the issue on my end.

**Comment by numinit** [comment]
Seems to fix it. While we're here, maybe we should clean up the Shellcheck warnings, and note why we should ignore the return value in a comment:

```
$ shellcheck --shell=bash build-support/setup-hooks/auto-patchelf.sh

In ../../../../build-support/setup-hooks/auto-patchelf.sh line 14:
addEnvHooks "$targetOffset" gatherLibraries
             ^-----------^ SC2154: targetOffset is referenced but not assigned.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 119:
        if [ -n "$runtimeDependencies" ]; then
                 ^------------------^ SC2154: runtimeDependencies is referenced but not assigned.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 132:
    local missing="$(
          ^-----^ SC2155: Declare and assign separately to avoid masking return values.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 178:
      $(find "$@" "${findOpts[@]}" \! -type d \
      ^-- SC2044: For loops over find output are fragile. Use find -exec or a while read loop.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 214:
      [ -n "$(echo "$segmentHeaders" | grep '^Program Headers:')" ] || continue
        ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ].


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 217:
          [ -n "$(echo "$segmentHeaders" | grep "^ *INTERP\\>")" ] || continue
            ^-- SC2143: Use grep -q instead of comparing output with [ -n .. ].


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 233:
    if [[ $depsMissing == 1 && -z "$autoPatchelfIgnoreMissingDeps" ]]; then
                                   ^----------------------------^ SC2154: autoPatchelfIgnoreMissingDeps is referenced but not assigned.


In ../../../../build-support/setup-hooks/auto-patchelf.sh line 247:
postFixupHooks+=('
                 ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.

For more information:
  https://www.shellcheck.net/wiki/SC2044 -- For loops over find output are fr...
  https://www.shellcheck.net/wiki/SC2154 -- autoPatchelfIgnoreMissingDeps is ...
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
```

**Comment by numinit** [comment]
All the SC2154 warnings look like they're all input variables, so those can probably be silenced FWIW.

**Comment by DavHau** [comment]
Fixed or silenced all red and orange shellcheck warnings. Please review

File: pkgs/build-support/setup-hooks/auto-patchelf.sh [hunk 1]
```diff
@@ -133,7 +139,7 @@ autoPatchelfFile() {
     missing="$(
         ldd "$toPatch" 2> /dev/null | \
             sed -n -e 's/^[\t ]*\([^ ]\+\) => not found.*/\1/p'
-    )"
+    )" || return 0
```
Comment (numinit) [inline pkgs/build-support/setup-hooks/auto-patchelf.sh:142]: Maybe a comment here on why ldd should fail?

**Comment by FRidh** [comment]
Let's keep this targeting master, rebuilds of packages that need patchelf are typically fast.

**Review by numinit — APPROVED** [review]
Overall looks good, a comment on ldd to explain why we're ignoring that return would be helpful before committing but I'm approving since others are blocked on this review.

**Comment by DavHau** [comment]
I added a comment explaining why we return on a failing ldd/sed command.

File: pkgs/build-support/setup-hooks/auto-patchelf.sh [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env bash
+
```
Comment (aszlig) [inline pkgs/build-support/setup-hooks/auto-patchelf.sh:2]: What's the reason for this? The file is not executable.

Comment (DavHau) [reply pkgs/build-support/setup-hooks/auto-patchelf.sh:2]: I added this to make shellcheck happy.

**Pull request merged**